### PR TITLE
[prometheus-pushgateway] - Add extra init containers for pushgateway helm chart

### PR DESCRIPTION
**Pull request by repl-mike-roest**
#### What this PR does / why we need it
This PR adds the ability for specifying init containers for the pushgateway helm chart.  We intend to use this to allow creating a DNS record pointing at our podIP to allow external consumers to use pushgateway without needing the cost of a ingress loadbalancer
#### Which issue this PR fixes
N/A
#### Special notes for your reviewer
This change was basically lifted directly from prometheus-blackbox-exporter as it has that support.
Testing output of pod with init container specified
```
Name:         test-pushgateway-prometheus-pushgateway-7fb84dc66f-j2qsq
Namespace:    do2
Priority:     0
Node:         ip-10-31-229-39.ca-central-1.compute.internal/10.31.229.39
Start Time:   Thu, 22 Sep 2022 09:24:41 -0600
Labels:       app=prometheus-pushgateway
              chart=prometheus-pushgateway-1.18.4
              heritage=Helm
              pod-template-hash=7fb84dc66f
              release=test-pushgateway
Annotations:  kubernetes.io/psp: eks.privileged
Status:       Running
IP:           10.31.233.82
IPs:
  IP:           10.31.233.82
Controlled By:  ReplicaSet/test-pushgateway-prometheus-pushgateway-7fb84dc66f
Init Containers:
  hello:
    Container ID:  containerd://14587246ce46ee41bcdf04577fcf0e604a815fd036c06b4ceb63bbd140275382
    Image:         alpine:3
    Image ID:      docker.io/library/alpine@sha256:bc41182d7ef5ffc53a40b044e725193bc10142a1243f395ee852a8d9730fc2ad
    Port:          <none>
    Host Port:     <none>
    Args:
      echo
      Done
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Thu, 22 Sep 2022 09:24:42 -0600
      Finished:     Thu, 22 Sep 2022 09:24:42 -0600
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-ll7ps (ro)
Containers:
  pushgateway:
    Container ID:   containerd://d9d36ed1d4c73fb7a2eb8f9323c0ce5f2be8a2da390edf37df5e06b1f662e17f
    Image:          prom/pushgateway:v1.4.2
    Image ID:       docker.io/prom/pushgateway@sha256:a684e7c830a4b19e564a93bfc3bf713e85b04ab9dfcab5633c14cbba241f9231
    Port:           9091/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Thu, 22 Sep 2022 09:24:45 -0600
    Ready:          True
    Restart Count:  0
    Liveness:       http-get http://:9091/-/ready delay=10s timeout=10s period=10s #success=1 #failure=3
    Readiness:      http-get http://:9091/-/ready delay=10s timeout=10s period=10s #success=1 #failure=3
    Environment:    <none>
    Mounts:
      /data from storage-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-ll7ps (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  storage-volume:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  kube-api-access-ll7ps:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  38s   default-scheduler  Successfully assigned do2/test-pushgateway-prometheus-pushgateway-7fb84dc66f-j2qsq to ip-10-31-229-39.ca-central-1.compute.internal
  Normal  Pulling    37s   kubelet            Pulling image "alpine:3"
  Normal  Pulled     37s   kubelet            Successfully pulled image "alpine:3" in 248.134657ms
  Normal  Created    37s   kubelet            Created container hello
  Normal  Started    37s   kubelet            Started container hello
  Normal  Pulling    36s   kubelet            Pulling image "prom/pushgateway:v1.4.2"
  Normal  Pulled     34s   kubelet            Successfully pulled image "prom/pushgateway:v1.4.2" in 1.794044824s
  Normal  Created    34s   kubelet            Created container pushgateway
  Normal  Started    34s   kubelet            Started container pushgateway
```
#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-pushgateway/templates/_helpers.tpl [hunk 1]
```diff
@@ -77,6 +77,10 @@ Returns pod spec
       imagePullSecrets:
 {{ toYaml .Values.imagePullSecrets | indent 8 }}
     {{- end }}
+      {{- if .Values.extraInitContainers }}
+      initContainers:
+{{ toYaml .Values.extraInitContainers | indent 8 }}
```
Comment (monotek) [inline charts/prometheus-pushgateway/templates/_helpers.tpl:82]:
```suggestion
        {{ toYaml .Values.extraInitContainers | nindent 8 }}      
```

File: charts/prometheus-pushgateway/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: "1.4.2"
 description: A Helm chart for prometheus pushgateway
 name: prometheus-pushgateway
-version: 1.18.3
+version: 1.18.4
```
Comment (monotek) [inline charts/prometheus-pushgateway/Chart.yaml:5]: new feature should raise minor version

```suggestion
version: 1.19.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-pushgateway/templates/_helpers.tpl (hunk 1 above)
Comment (repl-mike-roest) [reply charts/prometheus-pushgateway/templates/_helpers.tpl:82]: No where else in this file seems to be using nindent does it make sense to create a different pattern?

File: charts/prometheus-pushgateway/Chart.yaml (hunk 2 above)
Comment (repl-mike-roest) [reply charts/prometheus-pushgateway/Chart.yaml:5]: done

**Comment by repl-mike-roest** [comment]
@monotek @gianrubio anything I'm missing to allow this to move forward?  Thanks!

File: charts/prometheus-pushgateway/templates/_helpers.tpl (hunk 1 above)
Comment (zanhsieh) [inline charts/prometheus-pushgateway/templates/_helpers.tpl:82]:
```suggestion
      {{- with .Values.extraInitContainers }}
      initContainers:
        {{- toYaml . | nindent 8 }}
```

**Review by zanhsieh — COMMENTED** [review]
NR

File: charts/prometheus-pushgateway/templates/_helpers.tpl (hunk 1 above)
Comment (monotek) [reply charts/prometheus-pushgateway/templates/_helpers.tpl:82]: "nindent" seems to be the default nowdays used by "helm create" command.
Therefore i would stick to that default.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-pushgateway/templates/_helpers.tpl (hunk 1 above)
Comment (repl-mike-roest) [reply charts/prometheus-pushgateway/templates/_helpers.tpl]: Ok makes sense. Done

**Comment by repl-mike-roest** [comment]
@monotek done the changes

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-pushgateway/templates/_helpers.tpl [hunk 3]
```diff
@@ -77,6 +77,9 @@ Returns pod spec
       imagePullSecrets:
 {{ toYaml .Values.imagePullSecrets | indent 8 }}
     {{- end }}
+      {{- if .Values.extraInitContainers }}
+      initContainers:{{ toYaml .Values.extraInitContainers | nindent 8 }}
```
[the remark this answers is not in the document]
Comment (monotek) [reply charts/prometheus-pushgateway/templates/_helpers.tpl:81]: or do it like @zanhsieh requested it above

**Comment by repl-mike-roest** [comment]
@monotek ok put that new line back

**Review by monotek — APPROVED** [review]

**Pull request merged**